projects/core/src/lib/operators/dialogs/raster-kernel/raster-kernel.component.html

```diff
@@ -0,0 +1,116 @@
+        matrix kernel with weights or as specialized kernel types, where only the size is specified.
+    </p>
+    <p>You can find more information at <a href="https://docs.geoengine.io/operators/rasterkernel.html">docs.geoengine.io</a>.</p>
+</geoengine-dialog-help>
+
+<form [formGroup]="form" (ngSubmit)="add()" fxLayout="column" *ngIf="projectHasRasterLayers$ | async; else noDataInfo">
+    <div fxFlex class="container">
+        <geoengine-layer-selection [types]="RASTER_TYPE" formControlName="rasterLayer"></geoengine-layer-selection>
+        <geoengine-dialog-section-heading title="Configuration" subtitle="Specify the operator"></geoengine-dialog-section-heading>
+
+        <mat-form-field appearance="fill">
+            <mat-label>Kernel Type</mat-label>
+            <mat-select formControlName="kernelType">
+                <mat-option value="convolution">Convolution</mat-option>
+                <mat-option value="standardDeviation">Standard Deviation</mat-option>
+            </mat-select>
+        </mat-form-field>
+
+        <ng-container [ngSwitch]="form.controls.kernelType.value">
+            <ng-container *ngSwitchCase="'convolution'">
+                <div class="matrix-controls">
+                    <button mat-button color="primary" type="button" (click)="enlargeMatrix()">
+                        <mat-icon>add_circle_outline</mat-icon>
+                    </button>
+                    <button mat-button color="primary" type="button" (click)="smallenMatrix()">
+                        <mat-icon>remove_circle_outline</mat-icon>
+                    </button>
+                    <button mat-button color="primary" type="button" (click)="rotate90()">
+                        <mat-icon>refresh</mat-icon>
+                    </button>
+                    <button mat-button color="primary" type="button" [matMenuTriggerFor]="matrixMenu">
+                        <mat-icon>settings_suggest</mat-icon>
+                    </button>
+                    <mat-menu #matrixMenu="matMenu">
+                        <button mat-menu-item (click)="presetSobel()">Sobel</button>
+                        <button mat-menu-item (click)="presetGaussianBlur()">Gaussian blur 5x5</button>
+                        <button mat-menu-item (click)="presetMean()">Mean 3x3</button>
+                    </mat-menu>
+                </div>
+                <div class="matrix">
+                    <table>
+                        <tr *ngFor="let row of getMatrix(); index as r">
+                            <td *ngFor="let value of row; index as c">
+                                <mat-form-field appearance="fill">
+                                    <input
+                                        matInput
+                                        type="number"
+                                        [ngModel]="value"
+                                        (ngModelChange)="setMatrixValue(r, c, $event)"
```

Comment:
somehow I can't input multi-digit numbers. The field loses focus after the first digit

Reply:
fixed it in [44d384e](https://github.com/geo-engine/geoengine-ui/pull/476/commits/44d384e6d1ccbc3be47f29601f3b6b351c9d0ef5)